### DISCUSSION TITLE
PAR / REC + .nii to BIDS

### DISCUSSION THREAD

**Discussion by Freeklinders** [Donders-Institute/bidscoin#167] [Q&A]
Hi, 

I am trying to convert an old PAR/REC dataset into BIDS format using bidscoiner. 
However, the T1-w images are already in .nii format, and when I use the converter they are not present in the new, bids-formatted folders. Should this be the case? So yes, should I just copy them into am anat/ subfolder in the new BIDS folder? 

Thanks!

Best, 
Freek

**Comment by marcelzwiers** [comment]
That should be possible when you add the [nibabel2bids](https://bidscoin.readthedocs.io/en/latest/plugins.html#nibabel2bids-a-generic-plugin-for-imaging-data) plugin. I am not sure if the bidseditor from the current stable version will work flawlessly with this because, in the development version, I fixed an issue that emerged when users were changing the datatype or suffix. I wouldn't copy things over yourself (or e.g. your scans tsv-files will get outdated)

**Reply by Freeklinders** [reply]
Hi Marcel, thanks a lot for your fast reply! 
So when running the bidsmapper I should do: 

bidsmapper [sourcefolder] [bidsfolder] --p nibabel2bids dcm2niix2bids 

So it uses both? 
Thanks!

**Comment by marcelzwiers** [comment]
Correct, This will run especially well if your nifti's were produced by
dcm2niix (in the past), i.e. when the have json sidecar files...

Op di 31 jan. 2023 om 16:17 schreef Freeklinders ***@***.***>:

**Comment by Freeklinders** [comment]
Dear Marcel, 

I managed to bidscoin all the files in my dataset, though I keep receiving the following errors for all my files when trying to validate the new bids folder; 
ERROR CODE 1: (NOT INCLUDED): 
Files with such naming scheme are not part of BIDS specification. 
ERROR CODE 2: ACQ_NAME_CONTAIN_ILLEGAL_CHARACTER
acq Name contain an Illegal Character hyphen or underscore. Please edit the filename as per BIDS spe
structure.

My (bids) folder structure is as follows (example field map):
BIDS_con_new/sub-1001/fmap/sub-1001_acq-WIPB0epiSENSE_magnitude_fieldmaphz.json

In the bids mapper all my files were green and I received the same error when I left out the whole acquisition field in the bids mapper (thereby getting rid of the capitals). 

Do you have any idea what I am doing wrong? 

Thanks a lot in advance!

Best,
Freek

**Comment by marcelzwiers** [comment]
Although the bids-validator says that the `acq` field is wrong, I suspect that what is really wrong is the `magnitude_fieldmaphz` suffix. Normally such suffix errors are caused by the plugins, e.g. by dcm2niix adding its own suffix on top of the BIDS suffix (`_magnitude` -> `_magnitude_fieldmaphz`). Which plugin produced this error?

**Comment by marcelzwiers** [comment]
I suspect you were converting PAR/RECs of a fieldmap (https://github.com/rordenlab/dcm2niix/blob/master/FILENAMING.md). I don't have much experience with PAR/REC, but the good news is that collaborators from the SPINOZA center may possibly improve this in the near future.

**Comment by marcelzwiers** [comment]
I can have a look if you could share that fieldmap data with me...

**Comment by Freeklinders** [comment]
Hi Marcel, 

Thank you for your fast reply. Indeed, I watched your presentation at the Spinoza last week via zoom, as I heard they also had problems with converting PAR/REC files (which, you were correct, I am converting as well). Unfortunately I am not in really close contact with them. 
What specific data of the field map do you need? 
Looking at the error, it seems that the bidsvalidator also produces is for the functional scans, e.g.: 

sub-1001_task-cuereactivity_acq-FEEPI_bold_e1.json

**Reply by Freeklinders** [reply]
This is the information in the PAR b0_epi file prior to bidscoin:

 Protocol name                      :   WIP B0_epi SENSE
.    Examination date/time              :   2016.02.11 / 18:38:34
.    Series Type                        :   Image   MRSERIES
.    Acquisition nr                     :   5
.    Reconstruction nr                  :   1
.    Scan Duration [sec]                :   103
.    Max. number of cardiac phases      :   1
.    Max. number of echoes              :   1
.    Max. number of slices/locations    :   128
.    Max. number of dynamics            :   1
.    Max. number of mixes               :   1
.    Patient position                   :   Head First Supine
.    Preparation direction              :   Right-Left
.    Technique                          :   T1TFE
.    Scan resolution  (x, y)            :   128  128
.    Scan mode                          :   3D
.    Repetition time [ms]               :   10.759 
.    FOV (ap,fh,rl) [mm]                :   256.000  256.000  208.000
.    Water Fat shift [pixels]           :   1.134
.    Angulation midslice(ap,fh,rl)[degr]:   -2.071  1.029  0.662
.    Off Centre midslice(ap,fh,rl) [mm] :   1.452  20.005  -1.256
.    Flow compensation <0=no 1=yes> ?   :   0
.    Presaturation     <0=no 1=yes> ?   :   0
.    Phase encoding velocity [cm/sec]   :   0.000000  0.000000  0.000000
.    MTC               <0=no 1=yes> ?   :   0
.    SPIR              <0=no 1=yes> ?   :   0
.    EPI factor        <0,1=no EPI>     :   1
.    Dynamic scan      <0=no 1=yes> ?   :   0
.    Diffusion         <0=no 1=yes> ?   :   0
.    Diffusion echo time [ms]           :   0.0000
.    Max. number of diffusion values    :   1
.    Max. number of gradient orients    :   1
.    Number of label types   <0=no ASL> :   0

From one such PAR/REC pair, bidscoin produces two .nii files, one magnitude (.magnitude) and one fieldmap (unit is Hz). The bids mapper recognizes the PAR file as magnitude (and that's what I set it to).

**Comment by marcelzwiers** [comment]
I have some dedicated code to deal with dcm2niix suffixes:
```python
# Rename all files that got additional postfixes from dcm2niix. See: https://github.com/rordenlab/dcm2niix/blob/master/FILENAMING.md
dcm2niixpostfixes = ('_c', '_i', '_Eq', '_real', '_imaginary', '_MoCo', '_t', '_Tilt', '_e', '_ph', '_ADC', '_fieldmaphz')      #_c%d, _e%d and _ph (and any combination of these in that order) are for multi-coil data, multi-echo data and phase data
[..]
```
But somehow this code is not picked up for PAR/REC data... It is very hard to see from a distance what is happening there, as the same seems to work for DICOM.

**Comment by marcelzwiers** [comment]
So the `.nii` files are ok, but the `.json` files are not?

**Reply by Freeklinders** [reply]
I must say I am completely new to BIDS converting, so I don't know exactly what _should_ be present in the .json files; the only thing I see is that they all contain this information (of course different per scan):
This is an example of the .json file of the associated .nii magnitude image after bidscoining it:

{
    "Manufacturer": "Philips",
    "PatientPosition": "HFS",
    "SeriesDescription": "ImageMRSERIES",
    "ProtocolName": "WIPB0_epiSENSE",
    "SeriesNumber": 5,
    "AcquisitionNumber": 5,
    "ImageComments": "AAT",
    "PhilipsRescaleSlope": 1.81197,
    "PhilipsRescaleIntercept": 0,
    "PhilipsScaleSlope": 0.00125247,
    "UsePhilipsFloatNotDisplayScaling": 1,
    "EchoTime": 0.00303,
    "RepetitionTime": 0.010781,
    "ImageOrientationPatientDICOM": [
        0.997885,
        0.0359316,
        -0.0541693,
        -0.0360676,
        0.999348,
        -0.00153503
    ],
    "ConversionSoftware": "dcm2niix",
    "ConversionSoftwareVersion": "v1.0.20220720",
    "IntendedFor": [
        "func/sub-1001_task-cuereactivity_acq-FEEPI_bold_e1.nii.gz",
        "func/sub-1001_task-cuereactivity_acq-FEEPI_bold_e2.nii.gz",
        "func/sub-1001_task-cuereactivity_acq-FEEPI_bold_e3.nii.gz"
    ]
}
